blog/2024-03-01-config-entry-reconfigure-step.md

```diff
@@ -0,0 +1,44 @@
+---
+author: G Johansson
+authorURL: https://github.com/gjohansson-ST
+title: "Config Entries can now provide a reconfigure step"
+---
+
+As of Home Assistant Core 2024.4 config entries can now be reconfigured by adding a `reconfigure` step in their config flows.s
+
+This is not to replace optional configuration (`OptionsFlow`) but instead to allow the user to change setup configuration after a config entry has been created.
+
+### Integrations updating existing config entries
+
+Some integrations implements a reconfiguration option by updating an existing config entry if a matching config entry is being created by the user. This is no longer allowed, integrations should be migrated to implement the `reconfigure` step.
```

Comment:
What should happen when we detect the user entering new data for an existing entry? What should happen with the updates parameter for the abort already configured method?

Reply:
I'm not sure exactly what you mean.
I think it's the norm to abort with "config entry already exist" so does this need further clarification here?

Reply:
There's an `updates` parameter for `_abort_if_unique_id_configured`. Should that parameter no longer be used? Should we deprecate that parameter if it's not allowed to use anymore?

Reply:
Good point. Didn't consider this (as I completely forgot that it existed).

It's quite some integrations which has this implemented but almost all of them are updating only the `host` ip so should be somewhat easy to replace with a `reconfigure` step.
I think we should go for this implementation in general e.g. deprecate the `updates` parameter of `_abort_if...` which can be done in a follow-up PR shortly if agreed.

Reply:
`_abort_if_unique_id_configured` should still be called with updates from discovery flows, so that parameter is still valid.
I don't think many integrations intentionally do updates from user flows, and I think those who do should stop doing that and instead implement a reconfigure flow. Maybe the abort message should be updated to hint at the reconfigure step?